### PR TITLE
Add docs references

### PR DESCRIPTION
**Pull request by sudojarvis**
## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [ ] Includes new or updated tests to cover the new feature
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/plots/densityplot.py [hunk 1]
```diff
@@ -107,13 +108,20 @@ def plot_density(
...
     show : bool, optional
         Call backend show function.
 
     Returns
     -------
     axes : matplotlib axes or bokeh figures
 
+    See Also
+    --------
+    plot_posterior :Plot Posterior densities in the style of John K. Kruschke’s book.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:123]: Add space between `plot_posterior :` and `Plot` to pass the checks. Similarly, between `plot_density :` and `Generate`.

**Comment by sudojarvis** [comment]
could you suggest how to create doc/readthrdocs

File: arviz/plots/densityplot.py [hunk 2]
```diff
@@ -107,13 +108,20 @@ def plot_density(
...
         Call backend show function.
 
     Returns
     -------
     axes : matplotlib axes or bokeh figures
 
+    See Also
+    --------
+    plot_posterior : Plot Posterior densities in the style of John K. Kruschke’s book. 
+    plot_density : Generate KDE plots for continuous variables and histograms for discrete ones.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:124]: Add an empty line space before `Examples`.

**Comment by roshnaeem** [comment]
> could you suggest how to create doc/readthrdocs

We need to pass this check for readthedocs preview. The guide to check the preview is available [here](https://github.com/arviz-devs/arviz/wiki/ArviZ-Hacktoberfest-2021#how-to-check-the-preview-of-docs).

File: arviz/plots/densityplot.py (hunk 2 above)
Comment (roshnaeem) [inline arviz/plots/densityplot.py:124]: Add an empty line before the `Examples`.

File: arviz/plots/densityplot.py [hunk 3]
```diff
@@ -107,13 +108,21 @@ def plot_density(
     backend_kwargs: bool, optional
         These are kwargs specific to the backend being used. For additional documentation
         check the plotting method of the backend.
+        These are kwargs specific to the backend being used, passed to :func:`matplotlib.pyplot.subplots`
+         or  :func:`bokeh.plotting.figure`.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:112]: `or` is indented. It should have the same indentation as other lines.

File: arviz/plots/densityplot.py [hunk 4]
```diff
@@ -43,13 +43,14 @@ def plot_density(
...
-        Any object that can be converted to an az.InferenceData object, or an Iterator returning
-        a sequence of such objects.
-        Refer to documentation of az.convert_to_dataset for details about such objects.
+        Any object that can be converted to an :class:`arviz.InferenceData` object, 
+        or an Iterator returning a sequence of such objects.
+        Refer to documentation of :func:`arviz.convert_to_dataset` for details about such objects.
     group: Optional[str]
-        Specifies which InferenceData group should be plotted.  Defaults to 'posterior'.
+        Specifies which class:`arviz.InferenceData` group should be plotted. 
+         Defaults to 'posterior'.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:51]: "Defaults to 'posterior'." should have the same indentation as other lines.

File: arviz/plots/densityplot.py [hunk 5]
```diff
@@ -43,13 +43,14 @@ def plot_density(
...
         Alternative values include 'prior' and any other strings used as dataset keys in the
-        InferenceData.
+        class:`arviz.InferenceData`.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:53]: ``class:`arviz.InferenceData` `` to  ``:class:`arviz.InferenceData` ``

File: arviz/plots/densityplot.py [hunk 6]
```diff
@@ -64,7 +65,7 @@ def plot_density(
         Defaults to 0.94.
     point_estimate : Optional[str]
         Plot point estimate per variable. Values should be 'mean', 'median', 'mode' or None.
-        Defaults to 'auto' i.e. it falls back to default set in rcParams.
+        Defaults to 'auto' i.e. it falls back to default set in `rcParams`.
```
Comment (roshnaeem) [inline arviz/plots/densityplot.py:68]: single backticks are used for italic text. Use double backticks for highlighting code keywords. ` ``rcParams`` `

**Comment by roshnaeem** [comment]
@sudojarvis, i guess you again removed the empty line before "Examples". That's why the `readthedocs` check is failing.

File: arviz/plots/densityplot.py [hunk 7]
```diff
@@ -107,13 +108,20 @@ def plot_density(
     backend_kwargs: bool, optional
         These are kwargs specific to the backend being used. For additional documentation
         check the plotting method of the backend.
+        These are kwargs specific to the backend being used, passed to
```
[the remark this answers is not in the document]
Comment (sudojarvis) [reply arviz/plots/densityplot.py:111]: sure

**Review by roshnaeem — APPROVED** [review]

**Comment by roshnaeem** [comment]
@sudojarvis, Thank you for this PR.  Feel free to work on other issues as well. Let me know if you want to work on some issue, I'll assign it to you.

**Comment by roshnaeem** [comment]
For future PRs, always create a new branch before making any changes, working in the main is not a good practice. Create a new branch for every new PR. See contributing guide. https://arviz-devs.github.io/arviz/contributing/contributing_prs.html#steps

**Comment by sudojarvis** [comment]
oh sorry this, I will create a new branch for every pr.

**Comment by roshnaeem** [comment]
Alos, it will be a good practice if you comment on the issue, so I'll assign the issue to you and no one else will start working on it.

**Comment by roshnaeem** [comment]
@sudojarvis, there are also other  `kwargs` in `plot_ess`, can you try to have a look, you need to follow the same process as it has been described for `backend_kwargs` [here](https://github.com/arviz-devs/arviz/wiki/ArviZ-Hacktoberfest-2021#adding-backend_kwargs). 
You can ask me if anything is confusing.

**Review by roshnaeem — APPROVED** [review]

**Comment by OriolAbril** [comment]
This PR seems to be modifying both plot_ess and plot_density, but plot_density also has a previous PR open: https://github.com/arviz-devs/arviz/pull/1838. Not completely sure what is the goal of the PR so I can't review properly. Please read the pull request template and either add a link to the issue (or issues) you plan to solve or a detailed description of the goal of the PR

**Pull request merged**